ffi/README.md

```diff
@@ -1,80 +1,39 @@
-# Experiments with Dart FFI
+# Dart FFI Samples
 
 A series of simple examples demonstrating how to call C libraries from Dart.
 
-Please note that the Dart FFI API is in active development and likely to change
-before release. This code is designed to work with *Dart version 2.5.0*.
+This code is designed to work with *Dart version 2.6.0* and above.
 
-## Instructions
+## Building native libraries
 
-The C make files are (currently) written to work on a Mac. To compile the
-libraries, go to the `c/` folder in each subproject and run:
+Each sample uses [CMake][cmake] to generate a Makefile. To build the native
+library for each sample:
 
-``` shell
-make dylib
+```bash
+cd hello_world/hello_library
+cmake .
+make
 ```
 
-A .dylib file should be created in the parent folder. Navigate to that and run
-the dart file.
+The `make` command creates a `libhello.dylib` (macOS), `libhello.dll`
+(Windows) or `libhello.so` (Linux) library file.
 
-On Windows:
+## Running
 
-A dll can be created in two ways.
-
-1. Using `gcc`
-
-    In the subproject's `c/` run
-
-    ```shell
-    make dll
-    ```
-
-    Install gcc from [msys2](https://www.msys2.org/), a more detailed [tutorial](https://github.com/orlp/dev-on-windows/wiki/Installing-GCC--&-MSYS2).
-
-    And add it to `path` eg: `C:\msys64\mingw64\bin\`.
-
-    And make sure `gcc -dumpmachine` gives something like `x86_64-w64-mingw32` but **NOT** `i686-w64-mingw32`
-    eg: `C:\msys64\mingw64\bin\gcc.exe`
-
-
-2. Using `cl` from the Visual Studio build.
-    Follow the [docs](https://docs.microsoft.com/en-us/cpp/build/building-on-the-command-line?view=vs-2019) to be able to run `nmake.exe` from cmd line.
-
-    * Visual Studio needs to be installed with c/c++ checked.
-    * Then add `\path\to\VC\Auxiliary\Build` to `PATH` eg: `C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Auxiliary\Build` 
-    * You can now run `vcvars32.bat` , `vcvars64.bat` , `vcvarsx86_amd64.bat` etc.from the command line as mentioned in the [docs](https://docs.microsoft.com/en-us/cpp/build/building-on-the-command-line?view=vs-2019#developer_command_file_locations).
-    * Then choose your setup and run `vcvars*.bat` and now, `cl.exe` , `nmake.exe` can be used from the command line.
-
-    In the subproject's `c/` run
-
-    ``` shell
-    nmake -f Nmakefile dll
-    ```
-
-    A .dll file should be created in the parent folder. Navigate to that and run
-    the dart file.
-
-    **Only applicable if on Windows:**
-
-    The c/c++ functions should be exported or dart will not be able to lookup the function definitions.
```

Comment:
@domesticmouse @johnpryan This is what I was referring to.

Reply:
I've raised (and self assigned) https://github.com/dart-lang/samples/issues/32 to confirm that this sample works as advertised on MS Windows.

Reply:
Awesome, thanks! :+1: